File: 0033_Search in Rotated Sorted Array/SearchinRotatedSortedArray.cs
public class Solution {
    public int Search(int[] nums, int target) {
        var l = 0;
        var r = nums.Length-1;
        
        while(l <= r)
        {
            var mid = (r-l)/2 + l;
            if(nums[mid] == target) return mid;
            if(nums[mid] >= nums[l])
            {
                if(target >= nums[l] && target < nums[mid])
                    r = mid-1;
                else
                    l = mid+1;
            }else{
                if(target > nums[mid] && target <= nums[r])
                    l = mid+1;
                else
                    r = mid-1;
            }
        }
        
        return -1;
    }
}

File: 0033_Search in Rotated Sorted Array/SearchinRotatedSortedArray.java
public class Solution {
    public int search(int[] nums, int target) {
        int l = 0;
        int r = nums.length - 1;
        while (l <= r){
            int mid = (r - l)/2 + l;
            if (nums[mid] == target)
                return mid;
        
            if (nums[l] <= nums[mid]){
                 if (nums[l] <= target && target < nums[mid]) 
                    r = mid - 1;
                 else
                    l = mid + 1;
            }else{
                if (nums[mid] < target && target <= nums[r])
                    l = mid + 1;
                 else
                    r = mid - 1;
            }
        }
        return -1;
    }
}